### ISSUE TITLE
Import should "pull in" topics instead of creating new ones where possible

### ISSUE THREAD

**Issue by devvmh** [metamaps/metamaps#704]
Importing has gotten fairly flexible, to the point where you at minimum only need a link or a name to create a new topic. The fields are:

```
id # used to match topic with synapses imported at the same time
name
link
desc # defaults to ''
x # defaults to using autolayout
y # defaults to using autolayout
permission # defaults to defer to map
metacode # defaults to Wildcard, or Reference if a link is provided
```

Matching topics to existing topics will be a bit of a challenge. x and y values can be safely ignored, but the other six fields (id, name, link, desc, permission, metacode) should match the pulled in topic to be pulled in.

#703 is another factor. If I copy-paste, I think it's best that topics are pulled in. This may confuse users, though, so I want us to keep thinking about this.

The trigger for a topic to be considered a link is that there is no name field, but there is a link field. In this case, if a topic exists with that link field, it should be pulled in. This may be annoying sometimes; we'll have to see how this feature works. It may be we need to create a boolean field on topics specifying if their link is meant to be the whole topic, or just an attachment. The model I'm proceeding with assumes that links generally are the whole topic. Once a "link topic" is in the database, though, the title is pulled from the web, so it's hard to differentiate from a random topic with the same link as an attachment.

If the topic pulled in by this method is public, I guess we should either a) not pull in and create instead or b) prompt the user.

There is also another exported field, user, which isn't currently used in export, but will need to be.

From writing this out, I've learned I need to do more UX thinking before implementing this. Comments welcome

**Comment by Bortseb** [comment]
All topics have a unique ID, even if they have the same title right? Also, our JSON export includes the UniqueIDs?

I am worried that if "pulling in" existing topics is our default, and not an intentional choice... then we run the risk of lots of long standing topics getting renamed or deleted by people who aren't aware that they brought in topics from other maps.

**Comment by devvmh** [comment]
This is true. i don't think this issue will be an easy one to resolve, because of all it entails
